### PR TITLE
DMS/Topics: add topics functionality for DMS

### LINKED ISSUE #366: [DMS] Implement kafkapremium topics support
Required for: https://github.com/opentelekomcloud/terraform-provider-opentelekomcloud/issues/1788

### PR DESCRIPTION
**Pull request by artem-lifshits**
### What this PR does / why we need it
This PR adds topics functionality for Kafka premium
### Which issue this PR fixes
Resolves: #366 

### Tests:
=== RUN   TestDmsList
--- PASS: TestDmsList (0.89s)
=== RUN   TestDmsLifeCycle
    instances_test.go:59: Attempting to create DMSv1 instance
    instances_test.go:96: DMSv1 instance successfully created: 90e5a591-aafb-4f59-a729-38919615028c
    instances_test.go:182: Attempting to create DMSv1 Topic
    instances_test.go:195: DMSv1 Topic successfully created: dms-topic-lqrx3lCT
    instances_test.go:201: Attempting to delete DMSv1 Topic
    instances_test.go:216: DMSv1 Topic successfully deleted
    instances_test.go:113: Attempting to update DMSv1 instance: 90e5a591-aafb-4f59-a729-38919615028c
    instances_test.go:123: DMSv1 instance updated successfully: 90e5a591-aafb-4f59-a729-38919615028c
    instances_test.go:102: Attempting to delete DMSv1 instance: 90e5a591-aafb-4f59-a729-38919615028c
    instances_test.go:109: DMSv1 instance deleted successfully: 90e5a591-aafb-4f59-a729-38919615028c
--- PASS: TestDmsLifeCycle (359.95s)
PASS

### CODE REVIEW

File: openstack/dms/v1/topics/requests.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+type CreateOptsBuilder interface {
+	ToTopicCreateMap() (map[string]interface{}, error)
+}
+
+type DeleteOptsBuilder interface {
+	ToTopicDeleteMap() (map[string]interface{}, error)
+}
+
+type CreateOpts struct {
+	Name             string `json:"id" required:"true"`
```
Comment (lego963) [inline openstack/dms/v1/topics/requests.go:16]: Name, but json ID 🤔

File: openstack/dms/v1/topics/results.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+type GetResult struct {
+	golangsdk.Result
+}
+
+type DeleteResult struct {
+	golangsdk.Result
+}
+
+type TopicName struct {
+	Name string `json:"id"`
```
Comment (lego963) [inline openstack/dms/v1/topics/results.go:18]: Same

File: openstack/dms/v1/topics/results.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	Partition        int    `json:"partition"`
+	Replication      int    `json:"replication"`
+	SyncReplication  string `json:"sync_replication"`
+	RetentionTime    int    `json:"retention_time"`
+	SyncMessageFlush string `json:"sync_message_flush"`
+}
+
+func (r CreateResult) Extract() (*TopicName, error) {
+	s := new(TopicName)
+	err := r.ExtractInto(s)
```
Comment (lego963) [inline openstack/dms/v1/topics/results.go]: ExtractIntoStructPtr

File: openstack/dms/v1/topics/results.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return s, nil
+}
+
+func (r GetResult) Extract() (*Topic, error) {
+	s := new(Topic)
+	err := r.ExtractInto(s)
```
Comment (lego963) [inline openstack/dms/v1/topics/results.go]: Same here

**Review by lego963 — CHANGES_REQUESTED** [review]

**Review by anton-sidelnikov — DISMISSED** [review]

File: openstack/dms/v1/topics/requests.go (hunk 1 above)
Comment (artem-lifshits) [reply openstack/dms/v1/topics/requests.go:16]: yeah documentation and API mentions ID while in reality topic name is used everywhere

Comment (anton-sidelnikov) [reply openstack/dms/v1/topics/requests.go:16]: image

File: openstack/dms/v1/topics/results.go (hunk 4 above)
Comment (artem-lifshits) [reply openstack/dms/v1/topics/results.go]: fixed

File: openstack/dms/v1/topics/results.go (hunk 3 above)
Comment (artem-lifshits) [reply openstack/dms/v1/topics/results.go]: fixed

**Review by lego963 — APPROVED** [review]

**Pull request merged**